internal/k8s/crdcontroller.go

```diff
@@ -671,7 +671,7 @@ func checkRefOnController(key, refKey, refValue string) error {
 
 	// For public clouds, check using network UUID in AWS, normal network API for GCP, skip altogether for Azure.
 	if lib.IsPublicCloud() && refModelMap[refKey] == "network" {
-		if lib.GetCloudType() == lib.CLOUD_AWS {
+		if lib.GetCloudType() == lib.CLOUD_AWS || lib.GetCloudType() == lib.CLOUD_OPENSTACK {
```

Comment:
instead of doing this all around, can we define some constant array, which has these values, and use that everywhere ?

Reply:
added UsesNetworkRef function to check for cloud type in lib